### PR TITLE
Add font rendering

### PR DESCRIPTION
**Pull request by IronicallySerious**
Needs #75 to merge first

### CODE REVIEW

File: rootex/framework/components/transform_component.cpp [hunk 1]
```diff
@@ -78,6 +78,26 @@ void TransformComponent::addTransform(const Matrix& applyTransform)
 	setTransform(getLocalTransform() * applyTransform);
 }
 
+JSON::json TransformComponent::getJSON() const
+{
+	JSON::json j;
```
Comment (murex971) [inline rootex/framework/components/transform_component.cpp]: Name json variable properly.

File: rootex/framework/components/hierarchy_component.cpp [hunk 2]
```diff
@@ -25,6 +25,30 @@ bool HierarchyComponent::removeChild(Ref<Entity> node)
...
+		j["parent"]["ID"] = 1;
+		j["parent"]["name"] = "Root";	
+	}
+	else
+	{
+		j["parent"]["ID"] = m_Parent->getID();
+		j["parent"]["name"] = m_Parent->getName();
+	}
+
+	JSON::json& jc = j["children"];
```
Comment (murex971) [inline rootex/framework/components/hierarchy_component.cpp]: 'jc' gives no idea about the variable. Rename it.

Comment (IronicallySerious) [reply rootex/framework/components/hierarchy_component.cpp]: 'j' for JSON and 'c' for the 'children' field in that JSON

File: rootex/framework/components/visual/diffuse_visual_component.cpp [hunk 3]
```diff
@@ -8,14 +8,18 @@
...
+Component* DiffuseVisualComponent::Create(const JSON::json& componentData)
 {
-	ImageResourceFile* imageRes = ResourceLoader::CreateImageResourceFile(componentData["texturePath"].tostring());
+	ImageResourceFile* imageRes = ResourceLoader::CreateImageResourceFile(componentData["texturePath"]);
 	Ref<Texture> texture(new Texture(imageRes));
 
 	Ref<DiffuseMaterial> material(new DiffuseMaterial(texture));
 
-	DiffuseVisualComponent* diffuseComponent = new DiffuseVisualComponent(RenderPass::Global, material, ResourceLoader::CreateVisualModelResourceFile(componentData["resFile"].tostring()));
+	VisualModelResourceFile* vmr = ResourceLoader::CreateVisualModelResourceFile(componentData["resFile"]);
```
Comment (murex971) [inline rootex/framework/components/visual/diffuse_visual_component.cpp]: Rename 'vmr'.

Comment (IronicallySerious) [reply rootex/framework/components/visual/diffuse_visual_component.cpp]: vmr is an acronym for Visual Model Resource. Also this variable is local scope only and is basically a shorthand for VisualModelResourceFile

File: rootex/framework/components/transform_component.cpp (hunk 1 above)
Comment (IronicallySerious) [reply rootex/framework/components/transform_component.cpp]: 'j' stands for JSON, inside every file getJSON() function in a child of Component

File: editor/editor.cpp [hunk 4]
```diff
@@ -9,6 +9,7 @@
 void Editor::initialize(HWND hWnd)
 {
 	BIND_EVENT_MEMBER_FUNCTION("EditorSaveAll", Editor::saveAll);
+	BIND_EVENT_MEMBER_FUNCTION("EditorAutosave", Editor::autoSave);
```
Comment (murex971) [inline editor/editor.cpp]: should be "EditorAuto**S**ave", IMO (according to the naming standards you are following.)

File: rootex/framework/components/visual/diffuse_visual_component.cpp (hunk 3 above)
Comment (murex971) [reply rootex/framework/components/visual/diffuse_visual_component.cpp]: Use of acronyms should be avoided, imo.

Comment (IronicallySerious) [reply rootex/framework/components/visual/diffuse_visual_component.cpp]: Using acronyms is okay in smaller scopes. This variable lasts only for 7 lines of code. This is manageable

**Pull request merged**